cli/src/commands/env.ts

```diff
@@ -0,0 +1,107 @@
+import Command from '../utils/command';
+import flags from '../utils/cli-flags';
+import { getEnvOrDie } from '../utils/helpers';
+import fromPairs from 'lodash.frompairs';
+
+function nonEmpty<T>(x: T[]) { return x.length > 0 }
+
+export default class Env extends Command {
+  public static description = 'manipulate environment of deployed app';
+
+  public static examples = [
+    `$ ${Command.cliBinName} env --list --name fluffy-teapot-76`,
+    `$ ${Command.cliBinName} env --get PGUSER --get PGPASSWORD`,
+    `$ PGPASSWORD=secret ${Command.cliBinName} env --set PGUSER=admin --set-from-env PGPASSWORD`,
+  ];
+
+  public static args = [];
+
+  public static flags = {
+    ...Command.flags,
+    'name': flags.string({
+      char: 'n',
+      description: 'If provided access variables for given app',
+    }),
+    'set': flags.keyValue({
+      char: 's',
+      description: 'set VAR=value ...',
+      multiple: true,
+      default: [],
+    }),
+    'unset': flags.string({
+      char: 'u',
+      description: 'unset VAR ...',
+      multiple: true,
+      default: [],
+    }),
+    'set-from-env': flags.string({
+      char: 'S',
+      description: 'set from value in local environment (more secure than --set)',
+      multiple: true,
+      default: [],
+    }),
+    'get': flags.string({
+      char: 'g',
+      description: 'get VAR',
+      multiple: true,
+      default: [],
+    }),
+    'list': flags.boolean({
+      char: 'l',
+      description: 'list all variables',
+      default: false,
+    }),
+  };
+
+  public async run() {
+    await this.authenticate();
+    const { flags: {
+      'name': appName,
+      get,
+      set,
+      'set-from-env': setFromEnv,
+      unset,
+      list,
+    } } = this.parse(Env);
+    const appId = await this.getAppIdByNameOrWorkingDirectory(appName);
+    // Various commands are *not* executed in sequence of specified
+    // flags, but this ordering makes sense.
+    if (nonEmpty(get)) {
+      await this.get(appId, get);
+    }
+    if (nonEmpty(set) || nonEmpty(setFromEnv) || nonEmpty(unset)) {
+      const setVariables = [
+        ...set.map(({ key, value }) => ({ variable: key, value })),
+        ...setFromEnv.map((variable) => ({ variable, value: getEnvOrDie(variable) })),
+      ];
+      await this.set(appId, setVariables, unset);
+    }
+    if (list) {
+      await this.list(appId);
+    }
+  }
+
+  protected async get(appId: string, variableNames: string[]) {
+    const { variables } = await this.lycanClient.getEnv(appId, variableNames);
+    const indexed = fromPairs(variables.map(({ variable, value }) => [variable, value]));
+    // Output in requested order.
+    for (const variable of variableNames) {
+      this.log(`${variable}=${indexed[variable]}`);
```

Comment:
Yes, it will (and I was expecting that and accepted it).  No, it cannot (and I was _not_ expecting that, and I'm not sure I accept it).
Suggestions?  I _could_ fail with an error, but that is probably not very nice -- you cannot see all the other variables.  OTOH, any string might be a value.

Reply:
Other commands like `list` have a `--format=json` flag which would be unambiguous when you really need it.

If only needed for debugging might be enough to make sure that values are shown when running with `DEBUG=*`.

Reply:
You can simply use escaping, for example with `${JSON.stringify(indexed[variable])}` to tell apart the string undefined from the value undefined, but it is a bit cumbersome